sed/core/workflow_recorder.py

```diff
@@ -27,6 +27,6 @@ def __init__(self, func):
 
     def __call__(self, *args, **kwargs):
         self.__class__.call_tracker.append(
-            MethodCall(self.func.__name__, args, kwargs),
+            MethodCall(self.func.__qualname__, args, kwargs),
         )
-        return self.func(*args, **kwargs)
+        return self.func(self, *args, **kwargs)
```

Comment:
Did you test that? I understand that this will pass the MethodCall instance instead of the one of the calling function.

At least when trying to instantiate the sed processor now, I get:

AttributeError: 'CallTracker' object has no attribute 'loader'

Reply:
Yes I did test that, with the same example I provided above.

You are not wrong but the update_wrapper argument basically makes the MethodCall instance into a Foo instance, from what I understood. I used it in the functional wrapper approach as well (but it uses a different function there).

I will try to get it to work with loader and other methods we have.

Reply:
@daviddoji if you have a better understanding of decorators, would be very helpful :)

Reply:
Never played too much with decorators (just silly ones than do timing). Maybe have a look at some examples in here: [https://wiki.python.org/moin/PythonDecoratorLibrary](https://wiki.python.org/moin/PythonDecoratorLibrary)

Reply:
I would also think that you should return self.func(*args,**kwargs) as the decorator should then just operate the function as normal. 
It seems to me that adding self adds an argument to the function before calling...

Reply:
Unfortunately that does not pass the self argument to the method, in the case where you don't provide all the arguments (default ones). This one worked for me for a test case but apparently isn't for the real situation. So definitely buggy and I need to understand better the handling.